src/Security/Authentication/Core/src/JsonDocumentAuthExtensions.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text.Json;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public static class JsonDocumentAuthExtensions
+    {
+        public static string GetString(this JsonDocument document, string key) =>
+            document.RootElement.GetString(key);
+
+        public static string GetString(this JsonElement element, string key) =>
+            element.TryGetProperty(key, out var property)
+                ? property.ToString() : null;
```

Comment:
`GetString()` not `ToString()`

If the `JsonElement` is not of `Type` `JsonValueType.String`, the results of `ToString()` are likely not what you want here (unless I am mistaken).

ToString() returns the entire sub-object for object/arrays and returns "true", "false" for bools, and "" for nulls.

Reply:
No, `.ToString()` is the equivalent of `.Value<string>()`. The destination for these values are represented as strings, the underlying data type is not significant. Yes it would be odd to get a full object but the caller usually knows their expected structure well enough to avoid that (or wants it to happen).

Reply:
Got it. Thanks for clarifying :)